core/trino-main/src/main/java/io/trino/sql/planner/LiteralInterpreter.java

```diff
@@ -144,6 +146,16 @@ protected Slice visitBinaryLiteral(BinaryLiteral node, Void context)
         @Override
         protected Object visitGenericLiteral(GenericLiteral node, Void context)
         {
+            // special case a common case
+            if (StandardTypes.BIGINT.equalsIgnoreCase(node.getType())) {
```

Comment:
This is super hacky. I don't think we should be adding special cases for specific types here.

My guess is that what's expensive in this method is the call to resolve the cast from BIGINT -> VARCHAR (due to the type inferencer being slow, etc). That is something that could be cached in this class so it doesn't keep getting resolved over and over for every literal.

Reply:
I agree a more general solution would be better.

In fact, we could cache both result type and operator.
However we would need to retrofit LiteralInterpreter from being a utility class into having some lifecycle.